src/scanner.zig

```diff
@@ -144,12 +146,38 @@ const Protocol = struct {
 
     fn parse(allocator: *mem.Allocator, parser: *xml.Parser) !Protocol {
         var name: ?[]const u8 = null;
+        var copyright: ?[]const u8 = null;
+        var toplevel_description: ?[]const u8 = null;
         var interfaces = std.ArrayList(Interface).init(allocator);
         while (parser.next()) |ev| switch (ev) {
             .open_tag => |tag| {
-                // TODO: parse copyright and description
-                if (mem.eql(u8, tag, "interface"))
+                if (mem.eql(u8, tag, "copyright")) {
+                    if (copyright != null)
+                        return error.DuplicateCopyright;
+                    if (parser.next()) |e| {
+                        if (e == .character_data) {
+                            copyright = try allocator.dupe(u8, e.character_data);
+                        } else {
+                            return error.BadCopyright;
+                        }
+                    } else {
+                        return error.UnexpectedEndOfFile;
+                    }
```

Comment:
I think this would be cleaner like so:
```zig
const e = parser.next() orelse return error.UnexpectedEndOfFile;
switch (e) {
    .character_data => |data| copyright = try allocator.dupe(u8, data),
    else => return error.BadCopyright,
}
```

Reply:
or even more concisely:
```zig
switch (parser.next() orelse break) {
    .character_data => |data| copyright = try allocator.dupe(u8, data),
    else => return error.BadCopyright,
}
```